### PR TITLE
fix: Remove wait(DOWNLOAD_DELAY) from checkLocalFile

### PR DESCRIPTION
**Pull request by CozyKourai**
- Remove delay before checking for downloads.
- Add a loop to check for file every 10ms. Error will be throw if the file still not exist after `DOWNLOAD_DELAY`
- increase `DOWNLOAD_DELAY` to avoid false error as we don't have to wait the whole delay anymore

### CODE REVIEW

**Comment by CozyKourai** [comment]
[Retirer le timeout du download](https://trello.com/c/XrGhnWHx/1991-retirer-le-timeout-du-download)

File: testcafe/tests/helpers/utils.js [hunk 1]
```diff
@@ -92,11 +92,22 @@ export const getLastExecutedCommand = ClientFunction(
...
+      logger.info(
+        `${filepath} exists on local drive. Waited for a total of ${i.toString()} microseconds`
+      )
+      return
+    }
+  }
+  //Will throw error if file was not download
+  await t
+    .expect(fs.existsSync(filepath))
+    .ok(`${filepath} doesn't exist after ${data.DOWNLOAD_DELAY} microseconds`)
```
Comment (Crash--) [inline testcafe/tests/helpers/utils.js]: hmm ok c'est quand l'assertion est bonne, non ? Vu le message, ça devrait être nok ?

Comment (CozyKourai) [reply testcafe/tests/helpers/utils.js]: `.ok()` vérifie que l'assertion est bonne en effet, mais testcafé n'affiche le message que lorsque le test a failed.

File: testcafe/tests/helpers/utils.js [hunk 2]
```diff
@@ -92,11 +92,22 @@ export const getLastExecutedCommand = ClientFunction(
...
-  await t.expect(fs.existsSync(filepath)).ok(`${filepath} doesn't exist`)
-  logger.debug(`${filepath} exists on local drive`)
+export async function checkLocalFile(t, filepath) {
+  for (var i = 0; i < data.DOWNLOAD_DELAY; i++) {
+    await t.wait(10)
+    if (fs.existsSync(filepath)) {
+      logger.info(
+        `${filepath} exists on local drive. Waited for a total of ${i.toString()} microseconds`
+      )
+      return
```
Comment (y-lohse) [inline testcafe/tests/helpers/utils.js]: Why `return`? Did you mean `break`?

File: testcafe/tests/helpers/utils.js [hunk 3]
```diff
@@ -92,11 +92,22 @@ export const getLastExecutedCommand = ClientFunction(
...
-export async function checkLocalFile(filepath) {
-  await t.wait(data.DOWNLOAD_DELAY)
-  await t.expect(fs.existsSync(filepath)).ok(`${filepath} doesn't exist`)
-  logger.debug(`${filepath} exists on local drive`)
+export async function checkLocalFile(t, filepath) {
+  for (var i = 0; i < data.DOWNLOAD_DELAY; i++) {
+    await t.wait(10)
+    if (fs.existsSync(filepath)) {
+      logger.info(
+        `${filepath} exists on local drive. Waited for a total of ${i.toString()} microseconds`
```
Comment (y-lohse) [inline testcafe/tests/helpers/utils.js]: I think it's milliseconds and not microseconds, but whatever. However be aware that the `fs.existsSync` also takes time, so with the current code we will be waiting for longer than `DOWNLOAD_DELAY`. I guess `DOWNLOAD__CHECK_RETRIES` or something like that might be more accurate.

File: testcafe/tests/helpers/utils.js (hunk 2 above)
Comment (CozyKourai) [reply testcafe/tests/helpers/utils.js]: Changed to `break`

File: testcafe/tests/helpers/utils.js (hunk 3 above)
Comment (CozyKourai) [reply testcafe/tests/helpers/utils.js]: `DOWNLOAD_DELAY` is now `DOWNLOAD_CHECK_RETRIES`

File: testcafe/tests/helpers/utils.js (hunk 2 above)
Comment (y-lohse) [reply testcafe/tests/helpers/utils.js]: Alright, let's see if the tests still pass after this change, since before they weren't running at all :/

Comment (Crash--) [reply testcafe/tests/helpers/utils.js]: ~~return it's the same as break in a for loop, no?~~

Oh no. You right! OMG

**Pull request merged**